sql-plugin/src/main/scala/com/nvidia/spark/rapids/NvtxRangeWithDoc.scala

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids
+
+import ai.rapids.cudf.{NvtxColor, NvtxRange}
+import java.io.{File, FileOutputStream}
+import scala.collection.mutable.ListBuffer
+
+sealed class NvtxId private(val name: String, val doc: String) {
+  def help(): Unit = println(s"$name|$doc")
+}
+
+object NvtxId {
+  val registeredRanges = new ListBuffer[NvtxId]()
+
+  private def register(nvtxId: NvtxId): Unit = registeredRanges += nvtxId
+
+  private def apply(name: String, doc: String): NvtxId = {
+    val ret = new NvtxId(name, doc)
+    register(ret)
+    ret
+  }
+
+  val ACQUIRE_GPU: NvtxId = NvtxId(name = "Acquire GPU", doc = "Time waiting for GPU semaphore " +
+    "to be acquired")
+
+  val RELEASE_GPU: NvtxId = NvtxId(name = "Release GPU", doc = "Releasing the GPU semaphore")
+}
+
+object NvtxRangeDocs {
+  def helpCommon(): Unit = {
+    println("---")
+    println("layout: page")
+    println("title: NVTX Ranges")
+    println("nav_order: 5")
+    println("parent: Developer Overview")
+    println("---")
+    println(s"<!-- Generated by NvtxRangeDocs.help. DO NOT EDIT! -->")
+    // scalastyle:off line.size.limit
+    println("""# RAPIDS Accelerator for Apache Spark Nvtx Range Glossary
+              |The following is the list of Nvtx ranges that are used throughout
+              |the plugin. To add your own Nvtx range to the code, create an NvtxId
+              |entry in NvtxRangeWithDoc.scala and create an `NvtxRangeWithDoc` in the
+              |code location that you want to cover, passing in the newly created NvtxId.
+              |
+              |See [nvtx_profiling.md](https://nvidia.github.io/spark-rapids/docs/dev/nvtx_profiling.html) for more info.
+              |
+              |""".stripMargin)
+    // scalastyle:on line.size.limit
+    println("\n## Nvtx Ranges\n")
+    println("Name | Description")
+    println("-----|-------------")
+  }
+
+  def main(args: Array[String]): Unit = {
+    val configs = new FileOutputStream(new File(args(0)))
+    Console.withOut(configs) {
+      Console.withErr(configs) {
+        helpCommon()
+        NvtxId.registeredRanges.foreach(_.help())
+      }
+    }
+  }
+}
+
+class NvtxRangeWithDoc(val id: NvtxId, color: NvtxColor) extends AutoCloseable {
```

Comment:
Can we please change the API? The `AutoClosable` was me trying to do an RAII like API, but that is not a great API for java/scala. I know this will require changes to the CUDF code to open up the API, but can we please add some static push and pop methods in NvtxRange that take the name and the color bits but do check if nvtx is enabled.

Then we can have each of the `NvtxId`s, might want to rename them, have an `apply` method so we get the following code instead.

```
Nntx.ACQUIRE_GPU(semWaitTimeNs) {
  f()
}
```

```
Nvtx.RELEASE_GPU {
  ...
}
```

You might have guessed from this that I would like the color to be a part of the `NvtxId` as well. I think it is better if we are consistent in the color for a given range and that we have it documented.